kore/src/Kore/Step/Axiom/Matcher.hs

```diff
@@ -185,6 +185,7 @@ matchOne pair =
     <|> matchInj         pair
     <|> matchOverload    pair
     <|> matchDefined     pair
+    <|> matchAnd         pair
```

Comment:
Shouldn't this be moved further up or it doesn't matter?

Reply:
I don't think it makes any difference because none of these cases can match the same terms, but myself I would move it up under `matchEqualHeads` because it's kind of a "primitive" operation.